thorlcr/activities/keyedjoin/thkeyedjoinslave-legacy.cpp

```diff
@@ -2418,11 +2418,43 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
         info.unknownRowsOutput = true;
     }
 
-    void serializeStats(MemoryBuffer &mb)
+    virtual void serializeStats(MemoryBuffer &mb) override
     {
-        CSlaveActivity::serializeStats(mb);
         ForEachItemIn(s, _statsArr)
-            mb.append(_statsArr.item(s));
+        {
+            unsigned __int64 v = _statsArr.item(s);
+            if (0 == v)
+                continue;
+            switch (s)
```

Comment:
simplify: better with a
constexpr StatisticKind mapping[] = { StNumIndexSeeks, StNumIndexScans, ... };
and index into that to provide the mapping.
stats.setStatistic(mapping[s]);